docs/FLOAT_OPERATIONS.md

```diff
@@ -0,0 +1,329 @@
+# XRPL WASM Float Operations
+
+## Overview
+
+This document describes the floating-point operations available to XRPL WebAssembly smart contracts and the planned migration from the current BigDecimal implementation to rippled's native Number class.
+
+The float operations in XRPL are specifically used for **fungible tokens** (also called IOUs), which is one of three amount types supported by the XRP Ledger.
+
+## Table of Contents
+
+- [XRPL Amount Types](#xrpl-amount-types)
+- [Current Architecture](#current-architecture)
+- [Fungible Token Float Format](#fungible-token-float-format)
+- [Available Operations](#available-operations)
+- [Migration Plan](#migration-plan)
+- [Implementation Details](#implementation-details)
+- [Testing Strategy](#testing-strategy)
+
+## XRPL Amount Types
+
+The XRPL Amount type (STAmount) can represent three different types of assets:
```

Comment:
please take a look at https://github.com/ripple/craft/blob/0fd81efff7a464a2a9569377fdea08d411237b2a/xrpl-std/src/core/types/amount/token_amount.rs#L14

The layout diagram plus the description below would be awesome.  

I wonder what we should do. Create a separate token.md file or XRPLAmountTypes.md file, dedicated to explain 3 types, or keep it here. Up to you